Automattic/wp-calypso
# Assembler: Use the navigation path to control the selected category and sub-panel

Related to #

## Proposed Changes

* Use the navigation path to control the selected category slug so that we don't need to keep the `selectedMainItem`, `selectedCategory`, and then we can use the selected category to control whether we should open the pattern list panel instead of using `isPanelOpen`.
* Introduce the `.pattern-assembler__sidebar-panel` container so the layout becomes as followed, and we don't need absolute positioning and margin-inline-start to control the layout
  ```
  -----------------------------------------
  | Sidebar | Sidebar Panel | Large Preview |
  -----------------------------------------
  ```
* Created the `ScreenPatternListPanel` component to display the pattern list as the sidebar panel
* Removed the `EmptyPattern` component as it's no longer in use.
* Since we are using the navigation path to control the selected category, we can support the query string called `initial_path` to open the pattern list panel by default in the future

Any thoughts about using the navigation path to control the selected category slug and the new layout with the sidebar panel?

## Testing Instructions



* Go to Theme Showcase
* Scroll down to select the Assembler CTA
* On the Assembler screen, ensure everything works as expected

## Pre-merge Checklist
